solr/solrj-zookeeper/build.gradle

```diff
@@ -37,7 +37,7 @@ dependencies {
     implementation 'org.apache.httpcomponents:httpclient'
     implementation 'org.apache.httpcomponents:httpcore'
 
-    implementation('org.apache.zookeeper:zookeeper', {
```

Comment:
Why was this done? It suggests any consumer of solrj-zookeeper would likely want to use ZK APIs directly.  But I don't think that's true.

Reply:
It doesn't necessarily mean that, it means that the users would likely want to use ZK classes. This is true because in particular because of `Stat`, `KeeperException` and `CreateMode`.

Why I added it in this PR? I'm not really sure.

Reply:
Wouldn't a user pretty be looking only for only ZkClientClusterStateProvider and even then only indirectly through SolrJ's reflection instantiation of it as a result of passing ZK URLs to CloudSolrClient.Builder?  I understand the number of public classes in basically any module is large so there indeed exists some interfaces of miscellaneous transitive dependencies.  But in practice, I think it's best to be conservative with class path exposure based on typical use (subjective, I know).  They can always add it if they need to.